src/node_dlopen.cc

```diff
@@ -0,0 +1,118 @@
+#include "node.h"
+#include "node_buffer.h"
+#include "v8.h"
+#include "env.h"
+#include "env-inl.h"
+
+namespace node {
+namespace dlopen {
+
+using v8::Boolean;
+using v8::Context;
+using v8::FunctionCallbackInfo;
+using v8::Handle;
+using v8::Integer;
+using v8::Local;
+using v8::Number;
+using v8::Object;
+using v8::String;
+using v8::Value;
+using v8::Uint32;
+
+
+static void Dlopen(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  const char* filename;
+  if (args[0]->IsNull()) {
+    filename = nullptr;
+  } else if (args[0]->IsString()) {
+    node::Utf8Value name(env->isolate(), args[0]);
+    filename = *name;
+  } else {
+    return env->ThrowTypeError(
+        "expected a string filename or null as first argument"
+    );
+  }
+
+  if (!Buffer::HasInstance(args[1]))
+    return env->ThrowTypeError("expected a Buffer instance as second argument");
+
+  Local<Object> buf = args[1].As<Object>();
+  uv_lib_t* lib = reinterpret_cast<uv_lib_t*>(Buffer::Data(buf));
```

Comment:
Doing some analysis to get a better idea if this could be properly abstracted away.